src/IO/H5/SpectralIo.cpp

```diff
@@ -0,0 +1,52 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "SpectralIo.hpp"
+
+#include <hdf5.h>
+#include <ostream>
+#include <string>
+#include <vector>
+
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/Helpers.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "OpenGroup.hpp"
+#include "Utilities/MakeString.hpp"
+
+namespace h5_detail {
+
+const std::vector<Spectral::Basis> allowed_bases() noexcept {
```

Comment:
sorry what is `const` here? A `const` return type doesn't change anything from just return by value.

Related: can this be made into a `std::array`? I'm not sure, but it would eliminate an allocation if it could be

Reply:
Sorry the `const` was left over from when this was a variable and not a function.  It can be made into a `std::array` I'm pretty sure so I will change it.